src/test/java/com/google/cloud/genomics/dataflow/functions/JoinNonVariantSegmentsWithVariantsTest.java

```diff
@@ -163,14 +163,14 @@ public void testBinVariantsFn() {
         DoFnTester.of(new JoinNonVariantSegmentsWithVariants.BinShuffleAndCombineTransform.BinVariantsFn());
 
     List<KV<KV<String, Long>, Variant>> binVariantsOutput = binVariantsFn.processBatch(input);
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200L), snp1)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200L), snp2)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200L), insert)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 199L), blockRecord1)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200L), blockRecord1)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 201L), blockRecord1)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 202L), blockRecord1)));
-    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200L), blockRecord2)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200000L), snp1)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200000L), snp2)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200000L), insert)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 199000L), blockRecord1)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200000L), blockRecord1)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 201000L), blockRecord1)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 202000L), blockRecord1)));
+    assertThat(binVariantsOutput, CoreMatchers.hasItem(KV.of(KV.of("chr7", 200000L), blockRecord2)));
```

Comment:
Thank you for the clarification, and that more documentation will be added with a future PR.  Of course I agree it doesn't make sense to merge variants on different chromosomes, but to double-check that the reference is not included in the argument list of the called `merger.merge()` function, I added printouts to `JoinNonVariantSegmentsWithVariants`.  Below is the output, and it does not seem to be collecting the reference information as a way to filter by, but is part of the `Variant` object that would need to be filtered:

```
Running com.google.cloud.genomics.dataflow.functions.JoinNonVariantSegmentsWithVariantsTest
The contents of `context.element().getKey().getValue()` are = 199000

The contents of `context.element().getValue()` are = [reference_bases: "A"
calls {
  genotype: 0
  genotype: 0
  call_set_name: "hom sample"
}
calls {
  genotype: -1
  genotype: -1
  call_set_name: "no call sample"
}
end: 202050
reference_name: "chr7"
start: 199005
]
...
```

What I am proposing are only simplifications to the way things are approached because they will not only be harder to debug later on, but more importantly expanded by users to increase the set of features available to this nice API, and other dependent APIs of Google Genomics.